### PR TITLE
clarify how multiple layers of .attrs work

### PR DESCRIPTION
**Pull request by vzaidman**
clarify how multiple layers of .attrs work to prevent confusions like in this issue: https://github.com/styled-components/styled-components/issues/3178

### CODE REVIEW

File: sections/basics/attaching-additional-props.md [hunk 1]
```diff
@@ -34,3 +34,38 @@ render(
 ```
 
 As you can see, we get access to our newly created props in the interpolations, and the `type` attribute is passed down to the element.
+
+### Overriding .attrs
+Notice that when wrapping styled components, `.attrs` are applied from the innermost styled component to the uppermost styled component.
```
Comment (probablyup) [inline sections/basics/attaching-additional-props.md:39]:
```suggestion
Notice that when wrapping styled components, `.attrs` are applied from the innermost styled component to the outermost styled component.
```

File: sections/basics/attaching-additional-props.md [hunk 2]
```diff
@@ -34,3 +34,38 @@ render(
...
+
+This allows each level to **override** lower levels of `.attrs`, similarly to how css properties override css properties of inner components.
```
Comment (probablyup) [inline sections/basics/attaching-additional-props.md:41]:
```suggestion
This allows each wrapper to **override** nested uses of `.attrs`, similarly to how css properties defined later in a stylesheet override previous declarations.
```

File: sections/basics/attaching-additional-props.md [hunk 3]
```diff
@@ -34,3 +34,38 @@ render(
...
+
+Here `Input`'s `.attrs` are applied first, and then `PasswordInput`'s `.attrs`:
```
Comment (probablyup) [inline sections/basics/attaching-additional-props.md:43]:
```suggestion
`Input`'s `.attrs` are applied first, and then `PasswordInput`'s `.attrs`:
```

**Review by probablyup — CHANGES_REQUESTED** [review]

**Comment by vzaidman** [comment]
took all suggestions. thanks!

**Review by probablyup — APPROVED** [review]
Thank you for the improvement!

**Pull request merged**